android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java

```diff
@@ -99,10 +101,88 @@
         return accountTestData;
     }
 
+    @Test
+    public void test_compare() throws InvalidUriException, SerializationException {
```

Comment:
I think it might be helpful to break this test into multiple unit tests. Right now it's testing the "equals" method in five scenarios. I think it's best to keep unit tests as small as possible, and to only invoke the method under test once. This helps the reader understand what the test is about.

That said, consider splitting this test up into multiple unit tests. One for each time equals or equivalent is called. WDYT?

Reply:
I think that's a good idea. Almost all of the tests have multiple use-cases in them, there are some pros and cons, let's discuss this offline and add a refactoring ticket for what we decide.